boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,346 @@
+    uint16_t *         tx_data,
+    uint16_t *         rx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin);
+
+static void Io_Efuse_CalculateParityBit(uint16_t *serial_data_input)
+{
+    uint16_t _serial_data_input = *serial_data_input;
+    bool     parity_bit;
+
+    for (parity_bit = 0; _serial_data_input > 0; _serial_data_input >>= 1)
+    {
+        parity_bit ^= READ_BIT(_serial_data_input, 1U);
+    }
+
+    if (parity_bit)
+    {
+        SET_BIT(*serial_data_input, PARITY_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(*serial_data_input, PARITY_BIT);
+    }
+}
+
+static ExitCode Io_Efuse_WriteToEfuse(
+    uint16_t *         tx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin)
+{
+    HAL_StatusTypeDef status = HAL_OK;
+
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_RESET);
+    status = HAL_SPI_TransmitReceive(
+        efuse_spi_handle, (uint8_t *)tx_data, (uint8_t *)tx_data, 1U, 100U);
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_SET);
+
+    if (status != HAL_OK)
+    {
+        return EXIT_CODE_TIMEOUT;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_ReadFromEfuse(
+    uint16_t *         tx_data,
+    uint16_t *         rx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin)
+{
+    // Send the command stored in tx_data to the status register, to read the
+    // data from the register address specified in tx_data
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_RESET);
+    HAL_StatusTypeDef status = HAL_SPI_TransmitReceive(
+        efuse_spi_handle, (uint8_t *)tx_data, (uint8_t *)rx_data, 1U, 100U);
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_SET);
+
+    if (status != HAL_OK)
+    {
+        return EXIT_CODE_TIMEOUT;
+    }
+
+    // Receive data from the register address specified in tx_data by sending
+    // dummy data
+    *tx_data = 0xFFFF;
+    // The data read from the register specified in tx_data is stored in rx_data
+    // after the SPI transfer
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_RESET);
+    status = HAL_SPI_TransmitReceive(
+        efuse_spi_handle, (uint8_t *)tx_data, (uint8_t *)rx_data, 1U, 100U);
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_SET);
+
+    if (status != HAL_OK)
+    {
+        return EXIT_CODE_TIMEOUT;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+struct Efuse_Context *Io_Efuse_Create(
+    SPI_HandleTypeDef *const spi_handle,
+    GPIO_TypeDef *           nss_port,
+    uint16_t                 nss_pin,
+    GPIO_TypeDef *           fsob_port,
+    uint16_t                 fsob_pin,
+    GPIO_TypeDef *           fsb_port,
+    uint16_t                 fsb_pin,
+    GPIO_TypeDef *           channel0_port,
```

Comment:
I'd call this channel_0_port/pin and channel_1_port/pin just to make things easier to read